src/isl/mod.rs

```diff
@@ -1,36 +1,76 @@
+//! Provides a way to construct ISL types/constraints programmatically into an internal representation.
+//!
+//! This module consists of three submodules that implement public facing API for constructing ISL types/constraint:
+//!
+//! * `isl_type` module represents a public facing schema type [IslType] which converts given ion content in the schema file
```

Comment:
This explanation covers a lot, but it's currently being exposed as part of the library's documentation (via the `//!` doc comment format). The information it contains straddles the line between being relevant to end users and being relevant to `ion-schema-rust` developers.

As a `TODO`, we should do one of the following:
1. Make this a developer comment (`//` instead of `//!`)
OR
2. Write the explanation with end users in mind as. In particular:
* Remove mentions of what's "public facing"
* Remove mentions of internal representations
* Remove mentions of how type resolution works
OR
3. Split it into two comments, one for the user's benefit and one for the maintainers.

My preference is number 3.